test/dash_analytics_test.dart

```diff
@@ -24,13 +29,16 @@ void main() {
   const String branch = 'branch';
   const String flutterVersion = 'flutterVersion';
   const String dartVersion = 'dartVersion';
+  const DevicePlatforms platform = DevicePlatforms.macos;
 
   setUp(() {
     // Setup the filesystem with the home directory
     fs = MemoryFileSystem.test();
-    home = fs.directory('home');
-    dartToolDirectory = home.childDirectory('.dart-tool');
+    home = fs.directory(homeDirName);
+    dartToolDirectory = home.childDirectory(kDartToolDirectoryName);
 
+    // The main analytics instance, other instances can be spawned within tests
```

Comment:
I thought by extending the default `AnalyticsImpl` class for `TestAnalytics` and just overriding the `sendEvent` method I was testing the package. Everything else is shared between the two classes... should `TestAnalytics` instead be implementing the abstract class?

Reply:
Good point, as discussed earlier today, I have captured this as an issue here https://github.com/eliasyishak/dash_analytics/issues/16 so that I can dedicate more time to working on mocking the client